### ISSUE TITLE
sensitive information logged?

### ISSUE THREAD

**Issue by Lanchon** [microg/android_packages_apps_GmsCore#251]
i have a google account configured. i am looking at my logcat, and i see stuff in the log that seems to be sensitive. in particular, i've seen suspect entries with these tags:
- GmsAuthManager
- GmsAuthManagerSvc
- GmsHttpFormClient

am i right? is there a way to disable this logging entries?

**Comment by mar-v-in** [comment]
You are right that microG logs several sensitive information, as does Android itself and other apps. Generally speaking, it's not a good idea to publish log files without understanding them first.

I agree that for certain portions of microG code that already runs pretty stable, it is a good idea to disable logging by default (by using android's `isLoggable()` mechanism that allows enabling logs by using setprop from adb)

**Comment by Lanchon** [comment]
i think it's logging the login cookies. IMHO that info shouldn't escape microG no matter what configuration is forced upon the app (unless you are root, of course). if you can alter the runtime environment to obtain the google cookies without being root, i'd say that is a microG vulnerability that needs fixing.

btw, does microG stores the account password or a derived hash on the phone? i suppose the password is typed into a webview and straight into google and microG never sees it, right?

and the possibility of a request for enhancement here: if microG uses a webview for google account login and/or creation, could it implement certificate pinning? this bypasses exploits of the PKI.

**Comment by ale5000-git** [comment]
I think that it may be ok to use android's isLoggable() mechanism for other sensitive informations but I think that for login cookies there must be a "compile time" settings disabled by default to log them (if they need to be logged at all for debugging) to avoid any security issue.
